connector/client/connector_client.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/infracloudio/msbotbuilder-go/schema"
+	"github.com/infracloudio/msbotbuilder-go/connector/auth"
+)
+
+type ConnectorClient struct {
+	Config ConnectorClientConfig
+}
+
+func (client *ConnectorClient) Post(target url.URL, activity schema.Activity) (int, error) {
+
+	token := client.getToken()
+
+	jsonStr, _ := json.Marshal(activity)
+	req, err := http.NewRequest("POST", target.String(), bytes.NewBuffer(jsonStr))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+token)
+
+	replyClient := &http.Client{}
+	resp, err := replyClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	return resp.StatusCode, err
+}
+
+func (client *ConnectorClient) getToken() string {
+
+	data := url.Values{}
+	data.Set("grant_type", "client_credentials")
```

Comment:
We can combine declaration and set values in single statement like -
```
data := url.Values {
        "grant_type":  "client_credentials",
}
```

Reply:
Actually the expected values are `[]string` and not `string`
So I think using Set() instead of instantiating multiple  `[]string` is better